### PR TITLE
SOLR-16996: Update Solr Exporter for Prometheus cli to use commons-cli

### PR DESCRIPTION
**Pull request by malliaridis**
https://issues.apache.org/jira/browse/SOLR-16996

# Description

This PR replaces the usage of `argparse4j` with `commons-cli` in `SolrExporter.java` of the `prometheus-exporter` module.

Note that the changes mark `baseUrl` as deprecated in favor to [SOLR-16824](https://issues.apache.org/jira/browse/SOLR-16824).

# Tests

I have executed the `prometheus-exporter` tests and `test_prometheus.bats` with Java 17 successfully. Running `test_prometheus.bats` with Java 21 seems to be problematic (even in current `main`).

# Checklist

- [X] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [X] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [X] I have developed this patch against the `main` branch.
- [X] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Review by epugh — COMMENTED** [review]
Lgtm...   so.. swapping parsers was possible without forcing any changes to the params?   It appears this would be totally back wards compatible?!!

**Comment by epugh** [comment]
Have you run the Prometheus export bats test?

**Comment by epugh** [comment]
I manually ran the bats test:  `./gradlew integrationTests --tests test_prometheus.bats` and am seeing:

```
> Task :solr:packaging:integrationTests
Running BATS tests with Solr base port 50749
1..1
# starting solr-exporter on 50849
not ok 1 should start solr prometheus exporter that scrapes solr for metrics in 15000ms
# (from function `assert_output' in file /Users/epugh/Documents/projects/solr/.gradle/node/packaging/node_modules/bats-assert/src/assert.bash, line 247,
#  in test file test/test_prometheus.bats, line 47)
#   `assert_output --partial 'solr_metrics_core_query_requests_total'' failed
# Waiting up to 180 seconds to see Solr running on port 50749 [|]  
# Started Solr server on port 50749 (pid=60534). Happy searching!
#
    
# -- output does not contain substring --
# substring (1 lines):
#   solr_metrics_core_query_requests_total
# output (4 lines):
#     % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
#                                    Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
#   curl: (7) Failed to connect to localhost port 50849 after 7 ms: Couldn't connect to server
# --
#
# Please find the SOLR_HOME snapshot for failed test #1 at: /Users/epugh/Documents/projects/solr/solr/packaging/build/test-output/failure-snapshots/1-1
# kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
# Last output:
#   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
#                                  Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
# curl: (7) Failed to connect to localhost port 50849 after 7 ms: Couldn't connect to server

> Task :solr:packaging:integrationTests FAILED
```

Could be a local issue...

**Comment by malliaridis** [comment]
> Lgtm... so.. swapping parsers was possible without forcing any changes to the params? It appears this would be totally back wards compatible?!!

Yes, both libraries do almost the same thing just in a different way.

> Have you run the Prometheus export bats test?

Thanks for reminding me. I only ran the java tests and didn't have the bats tests in mind. I misused the `OptionGroup` class which caused the bats tests to fail. Should work now as expected.

**Comment by epugh** [comment]
Okay, manually ran the BATS test after your changes and it passed.  I tweaked the github configuration to also run the bats tests when ANY of the exporter code changes.....   Maybe that is too frequent?  @janhoy may have a thought on this?

File: versions.props [hunk 1]
```diff
@@ -29,7 +29,6 @@ io.prometheus:*=0.16.0
 io.swagger.core.v3:*=2.2.22
 jakarta.ws.rs:jakarta.ws.rs-api=3.1.0
 junit:junit=4.13.2
-net.sourceforge.argparse4j:argparse4j=0.9.0
```
Comment (cpoerschke) [inline versions.props:32]: The corresponding `licenses` files can be removed now then too I think, opened #2627 for that.

**Pull request merged**